merino-web/src/endpoints/suggest.rs

```diff
@@ -48,29 +48,18 @@ async fn suggest(
         &query_parameters,
     );
 
-    let provider = provider
-        .get_or_try_init(settings.as_ref(), metrics_client.as_ref())
-        .await
-        .map_err(|error| {
-            tracing::error!(
-                ?error,
-                r#type = "web.suggest.setup-error",
-                "suggester error"
-            );
-            HandlerError::internal()
-        })?;
-
     let extensions = request.extensions();
     let request_id: &Uuid = extensions
         .get::<RequestId>()
         .ok_or_else(HandlerError::internal)?;
+    let provider_ref = &provider.clone().0;
```

Comment:
This isn't `provider_ref` it's `IdMulti` (the provider). Also the `Arc::clone` isn't needed here
```suggestion
    let provider = &provider.0;
```